build_tools/aws-sdk-code-generator/lib/aws-sdk-code-generator/views/spec/endpoint_provider_spec_class.rb

```diff
@@ -141,7 +141,7 @@ def transform_operation_values(value)
             case value
             when Hash
               value.each_with_object({}) do |(k, v), o|
-                o[Underscore.underscore(k)] = transform_operation_values(v)
+                o[Underscore.underscore(k).to_sym] = transform_operation_values(v)
```

Comment:
Does this need to be applied in the underscores in initialize too?

Reply:
Confusingly, no (which is why we didn't catch this issue on regular, non nested hashes).  In the view we do: 
```
          client.{{operation_name}}(
            {{#operation_params}}
            {{param}}: {{{value}}},
            {{/operation_params}}
          )
```

So the top level key should be a string since we do `{{param}}:`

Reply:
We should probably refactor this to not iterate operation_params and instead render operation_params as is so it's not so confusing.